tensorboard/components/tf_ng_tensorboard/settings/dialog.component.test.ts

```diff
@@ -166,4 +170,35 @@ describe('settings dialog test', () => {
       expect(dispatchSpy).not.toHaveBeenCalled();
     }));
   });
+
+  describe('changePageSize', () => {
+    it('dispatches changing the value', fakeAsync(async () => {
+      const fixture = TestBed.createComponent(SettingsDialogComponent);
+      fixture.detectChanges();
+
+      const reloadPeriod = fixture.debugElement.query(By.css('.page-size'));
+      reloadPeriod.nativeElement.value = 20;
+      reloadPeriod.nativeElement.dispatchEvent(new Event('input'));
+
+      expect(dispatchSpy).not.toHaveBeenCalled();
+
+      // We debounce it so it does not spam other components on very keystroke.
+      tick(500);
+
+      expect(dispatchSpy).toHaveBeenCalledWith(changePageSize({size: 20}));
+    }));
+
+    it('does not dispatch when input is invalid', fakeAsync(async () => {
+      const fixture = TestBed.createComponent(SettingsDialogComponent);
+      fixture.detectChanges();
+
+      const reloadPeriod = fixture.debugElement.query(By.css('.reload-period'));
+      reloadPeriod.nativeElement.value = 0;
+      reloadPeriod.nativeElement.dispatchEvent(new Event('input'));
+
+      tick(1e10);
```

Comment:
Wow!  Perhaps just `flush()` from '@angular/core/testing', to skip to the end, instead of using a large, magic number?

and if that works, on L186 also?